File: src/PackageManifestHelper/ManifestLocalization.cpp
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

#include "pch.h"
#include "yaml-cpp\yaml.h"
#include "ManifestLocalization.h"

namespace AppInstaller::Package::Manifest
{
    void ManifestLocalization::PopulateLocalizationFields(YAML::Node localizationNode)
    {
        // Required
        this->Language = localizationNode["Language"].as<std::string>();

        // Optional
        this->Description = localizationNode["Description"] ? localizationNode["Description"].as<std::string>() : "";
        this->Homepage = localizationNode["Homepage"] ? localizationNode["Homepage"].as<std::string>() : "";
        this->LicenseUrl = localizationNode["LicenseUrl"] ? localizationNode["LicenseUrl"].as<std::string>() : "";
    }
}

File: src/WinGetUtilInterop/Manifest/Preview/ManifestLocalization.cs
// -----------------------------------------------------------------------------
// <copyright file="ManifestLocalization.cs" company="Microsoft Corporation">
//     Copyright (c) Microsoft Corporation. Licensed under the MIT License.
// </copyright>
// -----------------------------------------------------------------------------

namespace Microsoft.WinGetUtil.Models.Preview
{
    /// <summary>
    /// Class that contains the elements that represent what a manifest localization element can contain in the manifest file.
    /// </summary>
    public class ManifestLocalization
    {
        /// <summary>
        /// Gets or sets the localization type of manifest. Required element.
        /// </summary>
        public string Language { get; set; }

        /// <summary>
        /// Gets or sets the description of the manifest.
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// Gets or sets the homepage related to the package.
        /// </summary>
        public string Homepage { get; set; }

        /// <summary>
        /// Gets or sets the license URL.
        /// </summary>
        public string LicenseUrl { get; set; }
    }
}
